### PR TITLE
Add task for data precollection

### PR DESCRIPTION
**Pull request by flying-pi**

### CODE REVIEW

File: rg_instructor_analytics/management/commands/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# flake8: noqa
```
Comment (idegtiarov) [inline rg_instructor_analytics/management/commands/__init__.py:1]: Why do we have this line in the init file?

File: rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""
+Manage.py command for the update precollected data.
```
Comment (idegtiarov) [inline rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py]: s/Manage.py command for the update precollected data./Manage.py command for pre-collecting analytics data./

File: rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+"""
+from django.core.management.base import BaseCommand
+
+from rg_instructor_analytics.tasks import run_common_static_collection
+
+
+class Command(BaseCommand):
+    """
+    Command for initial collect data for the rg_instructor_analytics.
```
Comment (idegtiarov) [inline rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py]: s/collect/collecting/

File: rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    """
+
+    help = """
+    Send task to celery for update internal data about platform.
```
Comment (idegtiarov) [inline rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py]: s/Send task to celery for update internal data about platform./Run Celery task to collect data required for analytics./

File: rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    Attention - collection of the initial data take several hours.
```
Comment (idegtiarov) [inline rg_instructor_analytics/management/commands/update_rg_analytics_precollection_data.py]: s/Attention - collection of the initial data take several hours./Attention - the first run of the collecting data may take several hours./

File: rg_instructor_analytics/tasks.py [hunk 6]
```diff
@@ -183,7 +184,10 @@ def get_grade_summary(user_id, course):
     """
     Return the grade for the given student in the addressed course.
     """
-    return CourseGradeFactory().create(User.objects.all().filter(id=user_id).first(), course).summary
+    try:
+        return CourseGradeFactory().create(User.objects.all().filter(id=user_id).first(), course).summary
+    except PermissionDenied:
```
Comment (idegtiarov) [inline rg_instructor_analytics/tasks.py:189]: Why does this exception could appear?

File: rg_instructor_analytics/tasks.py [hunk 7]
```diff
@@ -231,3 +237,12 @@ def grade_collector_stat():
             GradeStatistic.objects.update_or_create(**key_values)
 
         LastGradeStatUpdate(last_update=this_update_date).save()
+
+
+@task
+def run_common_static_collection():
+    """
+    Task for the initial data collection.
```
Comment (idegtiarov) [inline rg_instructor_analytics/tasks.py]: s/Task for the initial data collection./The task for updating analytics data./

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Some questions and minor changes are online please take a look.

File: rg_instructor_analytics/management/commands/__init__.py (hunk 1 above)
Comment (flying-pi) [reply rg_instructor_analytics/management/commands/__init__.py:1]: For avoiding error from flake8 about missing docstring for the public module.

File: rg_instructor_analytics/tasks.py (hunk 6 above)
Comment (flying-pi) [reply rg_instructor_analytics/tasks.py:189]: This code from the previous commit. It handle a situation when the user loses his access to the course block.

File: rg_instructor_analytics/tasks.py (hunk 7 above)
Comment (flying-pi) [reply rg_instructor_analytics/tasks.py]: Replace with `The task for updating analytics data.`. For your, variant flake8 generate error

**Review by idegtiarov — APPROVED** [review]

**Pull request merged**